### ISSUE TITLE
very low mAP when training from scratch and using bn

### ISSUE THREAD

**Issue by ZQ-zz** [open-mmlab/mmdetection#1468]
I got bad result when training from and using bn or syncBN.
I think I config is right for bn:
norm_cfg = dict(type='BN', requires_grad=True)
norm_eval = False
and change the norm_cfg in backbone, neck and head
case 1:
coco dataset from scratch
I train fcos, and change input size to (512, 512), batch size can up to 28 for single gpu, after 74epoch get only 18.9; but when using pretrained model and gn, can get mAP about 35.7.
case 2: 
own dataset from scratch
when I train my own dataset, after 10epoch, gn can get mAP about 21.7 and bn can only get mAP 2.0, the two config is the same except for bn and gn and batch size is 16 for one gpu.

I think when using bn and batch size is big enough, training from scratch can get similar result with pretrained result.  It seems there is some thing wrong with bn ？ Could you reproduce any model without pretrained weights and using bn or syncBN？ I have tried fcos，retinaNet，r50 and hrnet... in mmdetection, but all failed

**Comment by ZQ-zz** [comment]
@hellock help

**Comment by fighterstz** [comment]
Hi, I met the same problem. Have it been solved? @ZQ-zz @hellock

**Comment by yxNONG** [comment]
can you report the result use syncBN?


"Batch Normalization (BN) [20], the popular normalization method used to train modern networks, partially makes training detectors from scratch difﬁcult. Object detectors are typically trained with high resolution inputs, unlike image classiﬁers. This reduces batch sizes as constrained by memory, and small batch sizes severely degrade the accuracy of BN [19, 34, 48]. This issue can be circumvented if pre-training is used, because ﬁne-tuning can adopt the pretraining batch statistics as ﬁxed parameters [17]; however, freezing BN is invalid when training from scratch. We investigate two normalization strategies in recent works that help relieve the small batch issue:
(i) Group Normalization (GN) [48]: as a recently proposed alternative to BN, GN performs computation that is independent of the batch dimension. GN’s accuracy is insensitive to batch sizes [48].
(ii) SynchronizedBatch Normalization (SyncBN) [34, 27]: this is an implementation of BN [20] with batch statistics computed across multiple devices (GPUs). This increases the effective batch size for BN when using many GPUs, which avoids small batches.
Our experiments show that both GN and SyncBN can enable detection models to train from scratch. "                                                  --------------<Rethinking ImageNet Pre-training>

**Comment by weyoung0** [comment]
Hi, I met the same problem. Have it been solved? @ZQ-zz @hellock @hhaAndroid
